src/metabase/driver/sql/query_processor.clj

```diff
@@ -606,11 +606,17 @@
   [driver [_ & args]]
   (apply hx/call :concat (mapv (partial ->honeysql driver) args)))
 
+;; this pseudo-expression returns only the substring function name, to simplify driver implementations
+(defmethod ->honeysql [:sql ::substring-name] [_ _] :substring)
```

Comment:
This approach could help us eliminate code duplication in places where a driver only has a different SQL keyword here and there, capitalization, etc.

Reply:
I think the closest thing is `add-interval-honeysql-form`. This is slightly more precise. Maybe a new multimethod is better, but we should be able to abstract out things like this to reduce duplication in driver implementations. It can be reverted if it's too weird.

Reply:
For now I'm going to back out all of the driver changes, since this is now a schema + frontend fix.